### PR TITLE
feat(cli): add option --security-only to diff

### PR DESCRIPTION
**Pull request by BryanPan342**
Create an option to `cdk diff` that checks for **broadening** security changes.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by BryanPan342** [comment]
A set up PR to resolve #12748

The idea moving forward is to:
- Create an option in `code-pipeline.ApplicationStage` called `securityCheck`
  - This option will create a `CodeBuild` before the ApplicationStage runs that does the following
     - `npm install -g aws-cdk`
     - `cdk diff --security-check-only` (might need to add `-a ./Assembly-Xxx`)
  - If this exit code is 1, then we have security changes and pause/wait for manual approval

File: packages/aws-cdk/lib/cdk-toolkit.ts
Comment (rix0rrr) [inline packages/aws-cdk/lib/cdk-toolkit.ts:98]: +? :)

File: packages/aws-cdk/lib/cdk-toolkit.ts
Comment (rix0rrr) [inline packages/aws-cdk/lib/cdk-toolkit.ts:111]: Honestly -- good call, but if there's already an option for this, let's just reuse. We can always pass `cdk diff --security-check-only --fail`.

File: packages/aws-cdk/test/integ/cli/cli.integtest.ts
Comment (rix0rrr) [inline packages/aws-cdk/test/integ/cli/cli.integtest.ts:489]: This test is going to fail. It doesn't set up a difference at all, so what is it going to check we have differences in?

I have to say, I'm a little mystified looking at the other diff tests on what exactly they do, but we should copy the struture from there at least.

**Review by rix0rrr — CHANGES_REQUESTED** [review]
Almost there!

File: packages/aws-cdk/lib/cdk-toolkit.ts
Comment (BryanPan342) [reply packages/aws-cdk/lib/cdk-toolkit.ts:98]: Returns true, so I use the + operator to typecast to a number

File: packages/aws-cdk/lib/cdk-toolkit.ts
Comment (BryanPan342) [reply packages/aws-cdk/lib/cdk-toolkit.ts:111]: O I was thinking that it would be something a user could use as well?

And the --fail would add more friction?

File: packages/aws-cdk/test/integ/cli/cli.integtest.ts
Comment (BryanPan342) [reply packages/aws-cdk/test/integ/cli/cli.integtest.ts:489]: O snap.. well I tested this on an actual CDK repo and it works HAHA, but yes I'll make this better

Comment (BryanPan342) [reply packages/aws-cdk/test/integ/cli/cli.integtest.ts:489]: I took this from the cdk deploy security changes one

Comment (BryanPan342) [reply packages/aws-cdk/test/integ/cli/cli.integtest.ts:489]: Wait I also ran this by itself and it worked?

File: packages/aws-cdk/bin/cdk.ts
Comment (rix0rrr) [inline packages/aws-cdk/bin/cdk.ts:117]: Can we change this to:

* `--security-only`

Instead?

Or do you suppose it would read better to do:

* `--only-security` ?

**Review by rix0rrr — APPROVED** [review]

**Comment by BryanPan342** [comment]
@mergifyio refresh

**Pull request merged**
